app/controllers/admin/collections_controller.rb

```diff
@@ -81,6 +81,26 @@ def destroy
     end
   end
 
+  def representative
+    @collection&.representative
+  end
+  helper_method :representative
+
+  def representative_is_image?
+    representative.file.present? && representative&.content_type&.start_with?("image/")
+  end
+  helper_method :representative_is_image?
+
+  def representative_dimensions_correct?
+    [ representative.width.present?,
+      representative.height.present?,
+      representative.width == representative.height,
+      representative.width >= CollectionThumbAsset::COLLECTION_PAGE_THUMB_SIZE * 2,
+    ].all?
```

Comment:
Moving to a helper was probably a good idea. 

I don't think `all` adds any clarity here over using `&&`, and possibly is slightly less efficient -- `&&` "short circuits", so if one is false the rest won't be calculated, while this formulation with an array and `all?` will necessarily calculate them all. 

But I don't think it's likely to be signfiicant, I'll just go ahead and approve anyway.

Reply:
[You're correct](https://stackoverflow.com/questions/20674612/do-all-and-any-guarantee-short-circuit-evaluation)- all? does not short circuit.

Reply:
Right, the way ruby works there is no way for it to, the array has to be constructed first before `all?` can be called on it. Just a ruby language syntax/semantics issue, the code to create the array, and all it's members, will be executed before ruby even gets to `all?`.